backend/src/test/java/ca/mcgill/purposeful/features/ID022_collaborationRequestWithMessage.java

```diff
@@ -0,0 +1,196 @@
+    CollaborationRequestDTO collaborationRequestDTO = new CollaborationRequestDTO();
+    collaborationRequestDTO.setMessage(message);
+    collaborationRequestDTO.setAdditionalContact(additionalContact);
+    collaborationRequestDTO.setIdea(new IdeaDTO(ideaRepository.findIdeaById(idMap.get(ideaId))));
+
+    // Create the request header
+    HttpHeaders header = cucumberUtil.bearerAuthHeader(jwtToken);
+    header.setContentType(MediaType.APPLICATION_JSON);
+    header.setAccessControlAllowOrigin("*");
+
+    // Create the request entity
+    HttpEntity<?> requestEntity = new HttpEntity<>(collaborationRequestDTO, header);
+
+    // Send the request
+    var response =
+        client.exchange(
+            "/api/collaborationRequest",
+            HttpMethod.POST,
+            requestEntity,
+            CollaborationRequestDTO.class);
+
+    // Extract returned DTO
+    ObjectMapper mapper = new ObjectMapper();
+    this.returnedCollaborationRequest =
+        mapper.convertValue(response.getBody(), new TypeReference<CollaborationRequestDTO>() {});
+  }
+
+  @Then("the number of collaboration requests in the database should be {string}")
+  public void theNumberOfCollaborationRequestsInTheDatabaseShouldBe(String number) {
+    assertEquals(Integer.parseInt(number), collaborationRequestRepository.count());
+  }
+
+  @And("the following error message shall be returned {string}")
+  public void theFollowingErrorMessageShallBeReturned(String errorMessage) {
+    assertEquals(errorMessage, returnedErrorMessage);
+  }
+
+  @Given("I am logged in as the user with email {string} and password {string} \\(ID022)")
+  public void iAmLoggedInAsTheUserWithEmailAndPasswordID(String email, String password) {
+    HttpEntity<String> requestEntity =
+        new HttpEntity<>(cucumberUtil.basicAuthHeader(email, password));
+
+    // We don't save this response in the field because we don't need it later
+    // In this case we are testing whether the browse ideas response is correct so
+    // we only
+    // need the token
+    ResponseEntity<?> response =
+        client.exchange("/api/login", HttpMethod.POST, requestEntity, String.class);
+    Assertions.assertEquals(
+        HttpStatus.OK, response.getStatusCode()); // Making sure the login was successful
+    jwtToken = response.getBody().toString(); // Extract the token for future requests
+    assertNotNull(jwtToken); // Ensure the token is not null
+  }
+
+  @And(
+      "the collaboration request with idea_id {string} and user_id {string} shall exist in the system")
+  public void theFollowingCollaborationRequestWithIdea_idAndUser_idShallExistInTheSystem(
+      String ideaId, String userId) {
+    List<CollaborationRequest> list =
+        collaborationRequestRepository.findCollaborationRequestsByRequesterAndIdea(
+            regularUserRepository.findRegularUserByAppUser_Id(idMap.get(userId)),
+            ideaRepository.findIdeaById(idMap.get(ideaId)));
+    assertEquals(1, list.size());
```

Comment:
I think this assertion is good, so long as we know the findCollaborationRequestsByRequesterAndIdea is working - did we unit test that in the repo tests? If not, maybe we just add an assertion on the message of the request, or we add a quick test in the repo class

Reply:
![image](https://user-images.githubusercontent.com/66871059/227749732-5ed93f35-635e-4489-9d92-faa9f1ab5ddc.png)

This is in collaborationRepositoryTests, added it last week when I wrote my first service method.

Reply:
Legend 👑 thanks!